### PR TITLE
Add compression option for ssh tunnels

### PR DESCRIPTION
**Pull request by pivnicek**
Add compression option for ssh tunnels

### CODE REVIEW

File: src/Keboola/DbExtractor/Configuration/ConfigDefinition.php [hunk 1]
```diff
@@ -113,6 +113,7 @@ public function addSshNode()
                 ->scalarNode('remotePort')->end()
                 ->scalarNode('localPort')->end()
                 ->scalarNode('user')->end()
+                ->booleanNode('compression')->defaultValue(false)->end()
```
Comment (ErikZigo) [inline src/Keboola/DbExtractor/Configuration/ConfigDefinition.php]: MySQL extractor pouziva kompresi na urovni DB spojeni. Tam se ten parametr uz jmenuje `networkCompression`. Mohlo by to byt asi stejne?

https://github.com/keboola/db-extractor-mysql/blob/master/src/Keboola/DbExtractor/Configuration/MySQLConfigDefinition.php#L37

File: composer.json [hunk 2]
```diff
@@ -16,10 +16,10 @@
     "keboola/php-datatypes": "^3.2",
     "pimple/pimple": "^3.0",
     "monolog/monolog": "^1.17",
-    "symfony/process": "^3.0",
+    "symfony/process": "^4.0",
```
[the remark this answers is not in the document]
Comment (pivnicek) [reply composer.json]: bylo to potreba za composer, nemohl to updatnout bez

File: src/Keboola/DbExtractor/Configuration/ConfigDefinition.php (hunk 1 above)
Comment (pivnicek) [reply src/Keboola/DbExtractor/Configuration/ConfigDefinition.php]: ale ten mysql networkCompression je option DB connection a tohle je option ssh connection tak nejsou uplnej stejne ne?

**Comment by ErikZigo** [comment]
Ta komprese se tomu tunelu stejne nepreda. Tady se to vyfiltruje.

https://github.com/keboola/db-extractor-common/blob/42af01d4ddde5f5c9943a04313bc28ddc3fc15c0/src/Keboola/DbExtractor/Extractor/Extractor.php#L102-L109

File: src/Keboola/DbExtractor/Configuration/ConfigDefinition.php (hunk 1 above)
Comment (pivnicek) [reply src/Keboola/DbExtractor/Configuration/ConfigDefinition.php]: bude to mozny pro mysql user po tom pouzivat oba, ten db option i tohle do hromady?

**Comment by pivnicek** [comment]
uz mam jinej error, mmt

**Comment by pivnicek** [comment]
ok @ErikZigo , myslim se ze uz to ma byt ok

File: composer.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (ErikZigo) [reply composer.json]: Ta knihovna nema byt v composer.json imho vubec.

File: src/Keboola/DbExtractor/Configuration/ConfigDefinition.php (hunk 1 above)
Comment (ErikZigo) [reply src/Keboola/DbExtractor/Configuration/ConfigDefinition.php]: Ano tam se to pouziva v DB, ne v SSH configuraci. Ale je to taky DB extractor tak by ty parametry mohly byt pojmenovane stejne.

Comment (pivnicek) [reply src/Keboola/DbExtractor/Configuration/ConfigDefinition.php]: bohuzel nemuzem, v ssh-tunnel lib to se jmenuje `compression`.  muzem pouzivat stejne labelu v UIcko

File: composer.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (pivnicek) [reply composer.json]: ? nechapu, co myslis?  neni tady `.lock`

[the remark this answers is not in the document]
Comment (pivnicek) [reply composer.json]: ah, rozumim, sorry jsem debyl.  zkusim ho zmazat

File: src/Keboola/DbExtractor/Configuration/ConfigDefinition.php (hunk 1 above)
Comment (ErikZigo) [reply src/Keboola/DbExtractor/Configuration/ConfigDefinition.php]: treba?

```php
$sshConfig['compression'] = $sshConfig['networkCompression']
```

File: tests/CommonExtractorTest.php [hunk 3]
```diff
@@ -953,6 +953,43 @@ public function testNoRetryOnCsvError(): void
...
+        $config = $this->getConfigRow(self::DRIVER);
+        $config['parameters']['db']['ssh'] = [
+            'enabled' => true,
+            'keys' => [
+                '#private' => $this->getPrivateKey(self::DRIVER),
+                'public' => $this->getEnv(self::DRIVER, 'DB_SSH_KEY_PUBLIC'),
+            ],
+            'sshHost' => 'sshproxy',
+            'localPort' => '33066',
+            'compression' => true,
```
Comment (tomasfejfar) [inline tests/CommonExtractorTest.php]: isn't this supposed to be false? Of what does it test? I don't think this needs to be tested separately as disabled compression is the default (i.e. there is already a test that has compression key unset in config).

File: tests/CommonExtractorTest.php [hunk 4]
```diff
@@ -953,6 +953,43 @@ public function testNoRetryOnCsvError(): void
...
+        ];
+        $result = ($this->getApp($config))->run();
+        $this->assertExtractedData($this->dataDir . '/simple.csv', $result['imported']['outputTable']);
```
Comment (tomasfejfar) [inline tests/CommonExtractorTest.php]: and the assertion is different with the same settings except local port that's highly unexpected.

Comment (pivnicek) [reply tests/CommonExtractorTest.php]: This is testing config row, the first one tests standard config  (2 different config definitions)

**Comment by tomasfejfar** [comment]
I also wasn't very keen about updating the SSH configuration node in two places, but that's already fixed in v10, so hopefully will be solved soon. (ref. https://keboola.slack.com/archives/C035LGNK5/p1544460078092600)

**Comment by pivnicek** [comment]
Yes, I cringed when I did it knowing how it's so much better in 10 :-)

File: tests/CommonExtractorTest.php (hunk 4 above)
Comment (tomasfejfar) [reply tests/CommonExtractorTest.php]: and why it'S called 

``` 
testSshWithFalseCompressionConfigRow
-------------^
```

File: tests/CommonExtractorTest.php (hunk 3 above)
Comment (tomasfejfar) [reply tests/CommonExtractorTest.php]: i see the difference now:

        $config = $this->getConfigRow(self::DRIVER);

vs 

        $config = $this->getConfig(self::DRIVER);

It's just it has such weird test method name ;)

File: tests/CommonExtractorTest.php (hunk 4 above)
Comment (pivnicek) [reply tests/CommonExtractorTest.php]: omg, sorry:)  I thought I had removed that `False`, thanks!

**Comment by pivnicek** [comment]
tak https://github.com/keboola/kbc-ui/pull/2507#issuecomment-446161735 hodim to na on-hold

**Comment by MiroCillik** [comment]
Preco je to `on-hold`?

**Pull request merged**